### ISSUE TITLE
Bug: White Screen / Account Doesn't Add

### ISSUE THREAD

**Issue by greggslist** [Foundry376/Mailspring#1847]
Submitting a ticket I got an automated response so I'm reporting issue here. Looks like others are experiencing similar problems. As a customer who has paid for the annual upgrade I hope there's a fix for this soon...

I've added a Gmail account without any issues as you can see in this screenshot - https://www.screencast.com/t/wi5zbZLzIVj8

However, I also test and I am unable to add am @outlook.com and email account using the mailserver on a cPanel server to Mailspring. This email account is hosted through a server using cPanel and I've tested IMAP/SMTP connections successfully with this email account using other email clients; Shift, Kiwi, Outlook and Thunderbird.

**Is there a fix for this or is this a known bug and will it be fixed/addressed in the near future?**

**Note:** I'm using Windows version of Mailspring.

**Two Scenarios:**
**1)** When I attempt to add a non-Gmail to Mailspring it loads a blank white screen. This occurs when there are no other (existing) accounts added into Mailspring.
Video: https://www.screencast.com/t/i7ByYwTDlM

**2)** When I attempt to add the non-Gmail account with another email account already added into Mailspring. It loads as if the add was successful, but then the account doesn't get added into Mailspring.
Video: https://www.screencast.com/t/RogNYmiTNM

I've also uninstalled Mailspring and deleted the data in C:\Users\<You>\AppData\Roaming\Mailspring per your help article here - https://foundry376.zendesk.com/hc/en-us/articles/115002012491

**Comment by 20Web** [comment]
@greggslist Had the same issue, what worked for me (worth a try until a fix is out) was deleting Mailspring then downloading 1.7.2 from here: https://github.com/Foundry376/Mailspring/releases/tag/1.7.2

Add all your email accounts on 1.7.2 then run the updater to go from 1.7.2 -> 1.7.4

**Comment by greggslist** [comment]
@20Web Awesome, thanks. This worked!

**Comment by evaldsurtans** [comment]
when are you planning to fix it as I suppose someone gets paid for subscriptions right?

**Comment by jefflenz** [comment]
Same issue on Ubuntu 18.04.4 LTS

**Comment by dylanhiemstra** [comment]
Same issue Linux Mint 19.3

**Comment by pythonInRelay** [comment]
Temp solution. Install an older version located [here](https://github.com/Foundry376/Mailspring/releases/tag/1.7.2).

**Comment by dllz** [comment]
Still waiting for this to get fixed

**Comment by hajk1** [comment]
Same issue version 1.7.4-d54ff730
Ubuntu 18.04.2 LTS

**Comment by lucian303** [comment]
Same on OS X 10.16.6 
Mailspring Version 1.7.4 (1.7.4)

**Comment by campfred** [comment]
Confirming issue on Windows 10 1909 with Mailspring 1.7.4.

**Comment by xergs** [comment]
> @greggslist Had the same issue, what worked for me (worth a try until a fix is out) was deleting Mailspring then downloading 1.7.2 from here: https://github.com/Foundry376/Mailspring/releases/tag/1.7.2
> 
> Add all your email accounts on 1.7.2 then run the updater to go from 1.7.2 -> 1.7.4

   
I've used the @20Web solution, then I set up the email accounts that I needed, an then updated to the last version. It worked for me (I'm using Debian 10.3 OS).

**Comment by bboennemann** [comment]
Same on Linux Mint 19.3. Ideapad s340, Ryzen 5. I would love  to use mailspring.

**Comment by vlopp** [comment]
Same thing, 

```
rijndael@rd:~$ mailspring -v
1.7.4-d54ff730
```

```
Operating System: Kubuntu 19.10
KDE Plasma Version: 5.18.3
KDE Frameworks Version: 5.67.0
Qt Version: 5.12.4
Kernel Version: 5.3.0-42-generic
OS Type: 64-bit
Processors: 8 × Intel® Core™ i7-7700HQ CPU @ 2.80GHz
Memory: 15,5 GiB of RAM
```

**Comment by bboennemann** [comment]
> Same on Linux Mint 19.3. Ideapad s340, Ryzen 5. I would love to use mailspring.

BTW, downloading the older version as suggested by @pythonInRelay works!
Link here: [https://github.com/Foundry376/Mailspring/releases/tag/1.7.2](url)

**Comment by tarantelklient** [comment]
It's also not possible to add a non-gmail account AFTER you already setup mailspring with a gmail-account.
I got an "Finished" screen, but my account is still missing.. I'd like to use snap for autoupdate and version 1.7.2 is not there.. So a downgrade is not that easy for me :/

**Comment by imuxin** [comment]
ditto,
os: 5.3.0-42-generic #34~18.04.1-Ubuntu SMP Fri Feb 28 13:42:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
mailpring -v:
Gtk-Message: Failed to load module "canberra-gtk-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
1.7.4-d54ff730

**Comment by tennom** [comment]
Had the same issue with fresh install on Ubuntu18.04. When it goes to white blank, it returns this in terminal. 


![mailspring](https://user-images.githubusercontent.com/8739460/77719327-fd0ce800-701f-11ea-86fb-1b30eab16a1c.png)

**Comment by joao-pedrozo** [comment]
Same issue on Ubuntu 18.04 LTS

**Comment by Alexander-Huang** [comment]
> @greggslist Had the same issue, what worked for me (worth a try until a fix is out) was deleting Mailspring then downloading 1.7.2 from here: https://github.com/Foundry376/Mailspring/releases/tag/1.7.2
> 
> Add all your email accounts on 1.7.2 then run the updater to go from 1.7.2 -> 1.7.4

Good.What I thought is the same.

**Comment by campfred** [comment]
That's a working workaround.
Although, I'm still pretty sure we shouldn't need to do that in order to use the app. I could be mistaken but the experience should simply be to be able to install the app, setup our account(s) and off we go with using it.

**Comment by sudoman281** [comment]
@Kozzi11 fixed this issue in this pull request: https://github.com/Foundry376/Mailspring/pull/1876. It's already merged.

**Comment by restructor1** [comment]
Same problem. Windows 10.

**Comment by mwikala** [comment]
> @greggslist Had the same issue, what worked for me (worth a try until a fix is out) was deleting Mailspring then downloading 1.7.2 from here: https://github.com/Foundry376/Mailspring/releases/tag/1.7.2
> 
> Add all your email accounts on 1.7.2 then run the updater to go from 1.7.2 -> 1.7.4

Worked for me! Using Ubuntu 19.10

**Comment by zalameedi** [comment]
> 
> 
> @greggslist Had the same issue, what worked for me (worth a try until a fix is out) was deleting Mailspring then downloading 1.7.2 from here: https://github.com/Foundry376/Mailspring/releases/tag/1.7.2
> 
> Add all your email accounts on 1.7.2 then run the updater to go from 1.7.2 -> 1.7.4

Fixed the issue on Windows 10.

**Comment by aigoncharov** [comment]
It's fixed in 1.7.5. If you're on Ubuntu and you're using `snap` to install the app, try updating it from the `edge` channel.

**Comment by tayfunyasar** [comment]
i cant see any release here, Am i seeing wrong? https://github.com/Foundry376/Mailspring/releases

**Comment by campfred** [comment]
> i cant see any release here, Am i seeing wrong? https://github.com/Foundry376/Mailspring/releases

No, you're right. Latest release is 1.7.4. Confirmed by the green « Latest release » badge.

**Comment by aigoncharov** [comment]
It's available as a [snap](https://snapcraft.io/mailspring). Probably, it's an automated nightly build from `master` or something like it.

**Comment by campfred** [comment]
That could be it. I haven't looked into snaps yet. Fingers crossed it'll be available in the release tab soon so that I can keep using it instead of Outlook!

**Comment by dllz** [comment]
Can confirm it is available at the link provided. 

For anyone wondering how to access it on Linux if you are already using 1.7.4: 
sudo snap switch mailspring --edge
sudo snap refresh

**Comment by heaviss** [comment]
I've just updated to 1.7.5 and still have white screen when press "Update connection settings". Ubuntu 19.10.

Here is what I get in console:

```js
{ Error: Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z
    at AttributeDateTime.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/attributes/attribute-datetime.ts:15:13)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:115:40)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/account.ts:126:24)
    at Account.clone (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:82:38)
    at new Composed (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/decorators/create-page-for-form.tsx:36:39)
    at Ag (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:135:201)
    at $g (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:157:149)
    at eh (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:13)
    at ai (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:202:255)
    at bi (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:203:271)
    at Fi (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218:174)
    at Gi (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217:174)
    at ii (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:214:252)
    at xg (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:212:94)
    at Object.enqueueSetState (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133:223)
    at InjectedComponentSet.Component.setState (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373:16)
    at ComponentRegistry._componentUnlistener.mailspring_exports_1.ComponentRegistry.listen (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/components/injected-component-set.tsx:114:12)
    at EventEmitter.eventHandler (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:128:23)
    at EventEmitter.emit (events.js:187:15)
    at EventEmitter.emit (domain.js:442:20)
    at ComponentRegistry.trigger (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:141:26)
    at ComponentRegistry.triggerDebounced.underscore_1.default.debounce (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/registries/component-registry.ts:197:44)
    at later (/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/underscore/underscore.js:828:25)
  message:
   'Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z' } { url:
   '/snap/mailspring/443/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js',
  line: 217,
  column: 372,
  pluginIds: [ 'onboarding' ] }
```

**Comment by cevheri** [comment]
Linux Mint 19.3 Cinnamon

<pre><font color="#8AE234"><b>cevher@NT00782</b></font>:<font color="#729FCF"><b>~</b></font>$ mailspring
Running database migrations
App load time: 1720ms

Running Setup
{&quot;error&quot;:null}
{ TypeError: Cannot read property &apos;emailAddress&apos; of undefined
    at _AccountStore.addAccount (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/stores/account-store.ts:223:18)
    at OnboardingStore._onFinishAndAddAccount (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/onboarding-store.ts:129:20)
    at EventEmitter.eventHandler (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/reflux/src/PublisherMethods.js:36:22)
    at EventEmitter.emit (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/eventemitter3/index.js:72:35)
    at Function.trigger (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/reflux/src/PublisherMethods.js:52:26)
    at Object.functor [as finishAndAddAccount] (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/reflux/src/createAction.js:32:56)
    at onboarding_helpers_1.finalizeAndValidateAccount.then.validated (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/decorators/create-page-for-form.tsx:170:29)
  message: <font color="#4E9A06">&apos;Cannot read property \&apos;emailAddress\&apos; of undefined&apos;</font> } { pluginIds: [ <font color="#4E9A06">&apos;onboarding&apos;</font> ] }
^C<font color="#8AE234"><b>cevher@NT00782</b></font>:<font color="#729FCF"><b>~</b></font>$ mailspring
Running database migrations
App load time: 192ms
</pre>

**Comment by dmatob** [comment]
Same issue on Debian 10

**Comment by MrSunshyne** [comment]
Same issue on KDE Neon. 
Installed from Snap. trying edge channel next

**Comment by bengotow** [comment]
Hey folks—thanks for filing this, this is a pretty serious bug. It was reported in a handful of other issues and we shipped a fix for it yesterday in version 1.7.5. 

You can download the update from GitHub releases or from https://getmailspring.com/download, and the app should also auto-update if you have been using it. Sorry for the delay getting a fix out for this one, I appreciate everyone's help debugging and triaging the problem!

**Comment by CodeMouse92** [comment]
Hi all,

2020 prevented much development work on Mailspring, but rest assured, development has resumed!

Also, please bear in mind that Mailspring Pro covers the costs of the APIs and server the pro features rely on. A Mailspring Pro subscription does *not* promise premium support. This is still an independent project which is primarily maintained by one person (Ben) in addition to his full time job.

It appears that this issue has been resolved, so we are closing it. If the problem occurs again, please create a new thread in the appropriate category on our [Discourse](https://community.getmailspring.com).

We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

We hope to see you there soon!

-The Mailspring Team
